<!DOCTYPE html>
<html lang="en">
<head>
<title>JS and jQuery | JS errors</title>
<meta charset="utf-8">
<meta name="description" content="HTML5, js, jquery demos" />
<!-- server
<link rel="stylesheet" type="text/css" href="..stylesheets/main1.css"></link>
-->
<link rel="stylesheet" type="text/css" href="stylesheets/main1.css"></link>
<style type="text/css"></style>
<script src="https://code.jquery.com/jquery-2.1.1.js"></script>
<script src="https://code.jquery.com/ui/1.11.1/jquery-ui.js"></script>
<script type="text/javascript">
	$(document).ready(function(){
		
	}); //end doc ready
</script>
</head>
<body>
	<input type="button" id="submit" class="btn" value="Refreshness" onclick="location.reload();" />
	<br /><br />
  <h1>JS errors</h1>

  <ul><span class="ul_tops">JS errors throwing and catching</span>
	  <li>errors will happen</li>
	  <li>errors can be a result of programming, bad input, unexpected behavior</li>
  	<li>try statement lets you test a block of code to look for errors</li>
  	<li>catch statement lets you handle errors</li>
  	<li>throw statement let your create custom errors</li>
  	<li>finally statement lets you execute code, after try and catch, in any result</li>
  </ul>

  <ul><span class="ul_tops">try and catch example</span>
  	<li id="here">first error</li>
  	<li>the try section of the code has a typo</li>
  	<li>the catch code block see the error and executes a code block to handle it</li>
  	<li> try { the try code block goes here } catch(error) { code block that handles errors };</li>
  	<li>try - statement lets you define a code block to be tested for errors while it is being executed</li>
  	<li>catch - statement lets you define a code block to be executed if an error it caught in the try code block</li>
  </ul>

  <ul><span class="ul_tops">throw errors</span>
  	<li>normally JS stops when an error occurs and generate a message</li>
  	<li>this is known as JS throwing an error</li>
  </ul>

  <ul><span class="ul_tops">throw statements</span>
  	<li>throw statement lets you create a custom error</li>
  	<li>known as: throw an exception</li>
    <li>the exception can be a JS string, Number, or Object</li>
    <li>throw "this right here"; // throw a string</li>
    <li>throw 123; // throw a number</li>
    <li>combine try, catch, and throw and now you can create custom errors</li>
  </ul>

  <ul><span class="ul_tops">input error detection</span>
    <li>Give me a number (no funny business) between 3-9</li>
    <li>Type here: <input id="demo" type="number" min="3" max="9" step="1"><button type="button" onclick="func_a()">Go!</button></li>
    <li><span id="message"></span></li>
  </ul>

<script>
	//document.getElementById("").
	//first error
	try{ adalert("Welcome to Jurrasic Park"); }
	//try{ addalert("Welcome to Jurrasic Park"); }
	catch(error) { document.getElementById("here").innerHTML = error.message; }

	$(document).ready(function(){

	}); //end doc ready

  function func_a(){
    var message, x;
    message = document.getElementById("demo").value;
    try{
      if(x=="") throw "did you forget how to use a keyboard?";
      if(isNaN(x)) throw "not a number. use the number pad will ya?"
        x == Number(x);
      if(x<3) throw "low baller?";
      if (x>9) throw "we have a high roller over here!";
    }
    catch(error){
      message.innerHTML = "O.K. " + error;
    }
  };

</script>
</body>
</html>